src/Symfony/Component/Security/Guard/Tests/Firewall/GuardAuthenticationListenerTest.php

```diff
@@ -184,6 +204,45 @@ public function testHandleCatchesAuthenticationException()
         $listener->handle($this->event);
     }
 
+    /**
+     * @legacy
+     */
+    public function testLegacyInterfaceAvoidSupportsCall()
+    {
+        $authenticatorA = $this->getMock('Symfony\Component\Security\Guard\GuardAuthenticatorInterface');
+        $providerKey = 'my_firewall3';
+
+        $authenticatorA
+            ->expects($this->never())
+            ->method('supports');
```

Comment:
Yeah, we need to change that. The recent version of PHPUnit's mock package now warns when you try to mock invalid methods and we therefore fixed some tests this week. Can you point me to the code this test should cover?